greenguard/pipeline.py

```diff
@@ -333,93 +417,47 @@ def cross_validate(self, X=None, y=None, readings=None, turbines=None, params=No
-            if isinstance(value, np.integer):
-                value = int(value)
-
-            elif isinstance(value, np.floating):
-                value = float(value)
+        elif hyperparams is None:
+            hyperparams = {}
 
-            elif isinstance(value, np.ndarray):
-                value = value.tolist()
+        template_splits = self._generate_splits(template_name, target_times, readings, turbines)
+        return self._cross_validate(template_splits, hyperparams)
 
-            elif value == 'None':
-                value = None
-
-            params_tree[block][hyperparameter] = value
-
-        return params_tree
-
-    def _to_tuples(self, params_tree, tunable_keys):
-        param_tuples = defaultdict(dict)
-        for block_name, params in params_tree.items():
-            for param, value in params.items():
-                key = (block_name, param)
-                if key in tunable_keys:
-                    param_tuples[key] = 'None' if value is None else value
-
-        return param_tuples
+    @staticmethod
+    def _parse_params(param_details):
+        param_type = param_details['type']
+        param_details['type'] = 'str' if param_type == 'string' else param_type
 
-    def _get_tunables(self):
-        tunables = []
-        tunable_keys = []
-        for block_name, params in self._pipeline.get_tunable_hyperparameters().items():
-            for param_name, param_details in params.items():
-                key = (block_name, param_name)
-                param_type = param_details['type']
-                param_type = 'string' if param_type == 'str' else param_type
+        if param_details['type'] == 'bool':
+            param_details['range'] = [True, False]
+        else:
+            param_details['range'] = param_details.get('range') or param_details.get('values')
 
-                if param_type == 'bool':
-                    param_range = [True, False]
-                else:
-                    param_range = param_details.get('range') or param_details.get('values')
+        if 'default' not in param_details:
+            param_details['default'] = param_details['range'][0]
 
-                value = HyperParameter(param_type, param_range)
-                tunables.append((key, value))
-                tunable_keys.append(key)
+        return param_details
 
-        return tunables, tunable_keys
+    @classmethod
+    def _get_tunables(cls, templates):
+        pipelines = {name: MLPipeline(template) for name, template in templates.items()}
+        tunables = {}
 
-    def _get_tuner(self):
-        tunables, tunable_keys = self._get_tunables()
-        tuner = GP(tunables)
+        for pipeline_name, pipeline in pipelines.items():
+            pipeline_tunables = {}
+            for name, param_details in pipeline.get_tunable_hyperparameters(flat=True).items():
+                pipeline_tunables[name] = cls._parse_params(param_details)
```

Comment:
Why parsing the params?
I think the output of the `get_tunable_hyperparameters(flat=True)` call should be able to be passed right away into the `Tunable.from_dict` method.